# Import PANDAS Library

Notice that we are using pd as an abbreviation so each time we want to use the PANDAS library, we can simply type pd instead of pandas.

In [1]:
import pandas as pd
pd.options.display.float_format = '{:,.2f}'.format

# Import Dataset (either full dataset or simplified)

You can use either set of datafiles (full or simplified). In the past some students' computers could not handle the full dataset. If you worry that your computer will not handle the full dataset (or if you find your code is running very slow), use the simplified version. 

The simplified version was created by taking a sample of the full data; hence the simplified files have "Sample" added to the end of the file names. Some files were not large to begin with and didn't need to be reduced. Those files do not have Sample added to the end of the file name.

## Adjust the file address below to match your computer.

You will want to replace the begining of the address below (i.e., everything before OneDrive - Oregon State University/)

In [2]:
begInv = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/BegInvFINAL12312016Sample.csv")

In [3]:
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  onHand  Price   startDate  
0  750mL       8  12.99  2016-01-01  
1  750mL       7  10.99  2016-01-01

In [4]:
begInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  
0  750mL               8  12.99  2016-01-01  
1  750mL               7  10.99  2016-01-01

# Deliverable 1a

In [5]:
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8  12.99  2016-01-01             103.92  
1  750mL               7  10.99  2016-01-01              76.93

In [6]:
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.head(2)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
1               49917         838,079.59
2               52925         840,452.20

In [7]:
begInv_Store.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852       3,291,170.24
73             162551       3,142,497.36
67             158996       3,079,578.63
66             149314       2,973,033.90
76             140208       2,952,418.44
69             144255       2,946,726.65
38             114368       2,232,698.77
55             119641       2,001,263.66
50              94720       1,649,808.22
79              95330       1,503,149.48

In [8]:
begInv_Store.sort_values(by='InvCost01_01_2016').tail(10).sort_values(by='InvCost01_01_2016', ascending = False)
###another way to get the same answers###

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852       3,291,170.24
73             162551       3,142,497.36
67             158996       3,079,578.63
66             149314       2,973,033.90
76             140208       2,952,418.44
69             144255       2,946,726.65
38             114368       2,232,698.77
55             119641       2,001,263.66
50              94720       1,649,808.22
79              95330       1,503,149.48

### Store 34 held the highest value of inventory at the start of the year.

# Deliverable 1b

In [9]:
begInv_Brand = begInv[['Brand','01_01_2016_Qty','InvCost01_01_2016']].groupby('Brand').sum()
begInv_Brand.head(2)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
58                281           3,650.19
60                288           3,165.12

In [10]:
begInv_Brand.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
3545            14499         463,823.01
1233            12016         432,455.84
8068            15341         383,371.59
4261            15499         340,823.01
3858            13649         327,439.51
2753             4625         286,703.75
8082             9287         278,517.13
8680             7066         275,503.34
2589             6766         270,572.34
3876            14829         266,773.71

### Brand 3543 had the highest value of inventory held at the start of the year.

# Deliverable 2a

In [11]:
endInv = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/EndInvFINAL12312016.csv")

In [12]:
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  onHand  Price     endDate  
0  750mL      11  12.99  2016-12-31  
1  750mL       7  36.99  2016-12-31

In [13]:
endInv = endInv.rename(columns = {'onHand':'12_31_2016_Qty'})
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  
0  750mL              11  12.99  2016-12-31  
1  750mL               7  36.99  2016-12-31

In [14]:
endInv['InvCost12_31_2016'] = endInv['12_31_2016_Qty'] * endInv['Price']
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11  12.99  2016-12-31             142.89  
1  750mL               7  36.99  2016-12-31             258.93

In [15]:
endInv_Store = endInv[['Store','12_31_2016_Qty','InvCost12_31_2016']].groupby('Store').sum()
endInv_Store.head(2)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
1               79827       1,206,845.93
2               56671         850,884.06

In [16]:
endInv_Store.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
50             260717       4,887,260.68
73             164589       3,254,662.81
67             163765       3,076,114.82
34             145829       3,074,616.75
76             143866       2,975,945.18
69             150848       2,968,678.82
66             144579       2,860,504.99
74             166015       2,803,645.13
38             129397       2,463,906.85
55             125584       2,234,836.35

### Store 50 held the highest value of inventory at the end of the year.

# Deliverable 2b

In [17]:
endInv_Brand = endInv[['Brand','12_31_2016_Qty','InvCost12_31_2016']].groupby('Brand').sum()
endInv_Brand.head(2)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
58                385           5,001.15
60                146           1,604.54

In [18]:
endInv_Brand.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
1233            15047         526,494.53
3545            16770         502,932.30
2753             7849         470,861.51
8068            15608         366,631.92
3405            12268         355,649.32
4261            16769         351,981.31
2757            11603         336,370.97
2589             7922         300,956.78
1376            13180         276,648.20
2585            10487         272,557.13

### Brand 1233 had the highest value in inventory at the EOY.

# Deliverable 3

In [19]:
dfSuperMerge = pd.merge(left=begInv,right=endInv,how='outer',on=['InventoryId'])
dfSuperMerge.head()

InventoryId  Store_x        City_x  Brand_x  \
0  1_HARDERSFIELD_58     1.00  HARDERSFIELD    58.00   
1  1_HARDERSFIELD_60     1.00  HARDERSFIELD    60.00   
2  1_HARDERSFIELD_62     1.00  HARDERSFIELD    62.00   
3  1_HARDERSFIELD_63     1.00  HARDERSFIELD    63.00   
4  1_HARDERSFIELD_72     1.00  HARDERSFIELD    72.00   

                 Description_x Size_x  01_01_2016_Qty  Price_x   startDate  \
0  Gekkeikan Black & Gold Sake  750mL            8.00    12.99  2016-01-01   
1       Canadian Club 1858 VAP  750mL            7.00    10.99  2016-01-01   
2     Herradura Silver Tequila  750mL            6.00    36.99  2016-01-01   
3   Herradura Reposado Tequila  750mL            3.00    38.99  2016-01-01   
4         No. 3 London Dry Gin  750mL            6.00    34.99  2016-01-01   

   InvCost01_01_2016  Store_y        City_y  Brand_y  \
0             103.92     1.00  HARDERSFIELD    58.00   
1              76.93      NaN           NaN      NaN   
2             221.94     1.00  HARDERSFIELD    62.00   
3             116.97     1.00  HARDERSFIELD    63.00   
4             209.94     1.00  HARDERSFIELD    72.00   

                 Description_y Size_y  12_31_2016_Qty  Price_y     endDate  \
0  Gekkeikan Black & Gold Sake  750mL           11.00    12.99  2016-12-31   
1                          NaN    NaN             NaN      NaN         NaN   
2     Herradura Silver Tequila  750mL            7.00    36.99  2016-12-31   
3   Herradura Reposado Tequila  750mL            7.00    38.99  2016-12-31   
4         No. 3 London Dry Gin  750mL            4.00    34.99  2016-12-31   

   InvCost12_31_2016  
0             142.89  
1                NaN  
2             258.93  
3             272.93  
4             139.96

In [20]:
dfSuperMerge = dfSuperMerge.rename(columns = {'01_01_2016_Qty':'Beginning_Qty'})
dfSuperMerge = dfSuperMerge.rename(columns = {'InvCost01_01_2016':'Beginning_Inv_Cost'})
dfSuperMerge = dfSuperMerge.rename(columns = {'12_31_2016_Qty':'Ending_Qty'})
dfSuperMerge = dfSuperMerge.rename(columns = {'InvCost12_31_2016':'Ending_Inv_Cost'})

In [21]:
dfSuperMerge[['InventoryId','Beginning_Qty','Beginning_Inv_Cost','Ending_Qty','Ending_Inv_Cost']].fillna(0).head()

InventoryId  Beginning_Qty  Beginning_Inv_Cost  Ending_Qty  \
0  1_HARDERSFIELD_58           8.00              103.92       11.00   
1  1_HARDERSFIELD_60           7.00               76.93        0.00   
2  1_HARDERSFIELD_62           6.00              221.94        7.00   
3  1_HARDERSFIELD_63           3.00              116.97        7.00   
4  1_HARDERSFIELD_72           6.00              209.94        4.00   

   Ending_Inv_Cost  
0           142.89  
1             0.00  
2           258.93  
3           272.93  
4           139.96

# Deliverable 4a

In [22]:
dfPurchases = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/PurchasesFINAL12312016Sample.csv")

In [23]:
dfPurchases.head(2)

InventoryId  Store  Brand                   Description   Size  \
0  69_MOUNTMEND_8412     69   8412     Tequila Ocho Plata Fresno  750mL   
1  34_PITMERDEN_5215     34   5215  TGI Fridays Long Island Iced  1.75L   

   VendorNumber                   VendorName  PONumber      PODate  \
0           105  ALTAMAR BRANDS LLC               8124  2015-12-21   
1          4466  AMERICAN VINTAGE BEVERAGE        8137  2015-12-22   

  ReceivingDate InvoiceDate     PayDate  PurchasePrice  Quantity  Dollars  \
0    2016-01-02  2016-01-04  2016-02-16          35.71         6   214.26   
1    2016-01-02  2016-01-07  2016-02-21           9.41         5    47.05   

   Classification  
0               1  
1               1

In [24]:
dfPurchases["InvoiceDate"] = pd.to_datetime(dfPurchases["InvoiceDate"])

In [25]:
mask = (dfPurchases['InvoiceDate'] >= '2016/01/01') & (dfPurchases['InvoiceDate'] <= '2016/12/31')

In [26]:
dfPurchases.groupby(['VendorNumber','VendorName']).agg({'Dollars':['sum']}).sort_values(('Dollars', 'sum'), ascending=False).head(10)

Dollars
                                                   sum
VendorNumber VendorName                               
3960         DIAGEO NORTH AMERICA INC    21,315,299.75
4425         MARTIGNETTI COMPANIES       11,709,222.19
12546        JIM BEAM BRANDS COMPANY     10,232,521.65
17035        PERNOD RICARD USA           10,224,899.93
480          BACARDI USA INC              7,420,630.63
1392         CONSTELLATION BRANDS INC     6,584,032.72
1128         BROWN-FORMAN CORP            5,719,375.62
9165         ULTRA BEVERAGE COMPANY LLP   5,545,766.29
3252         E & J GALLO WINERY           5,177,067.61
9552         M S WALKER INC               4,647,908.17

# Deliverable 4b

In [27]:
dfInvoices = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/InvoicePurchases12312016.csv")

In [28]:
dfInvoices.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6   214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15   140.55     8.57      NaN

In [29]:
dfInvoices["InvoiceDate"] = pd.to_datetime(dfInvoices["InvoiceDate"])

In [30]:
mask = (dfInvoices['InvoiceDate'] >= '2016/01/01') & (dfInvoices['InvoiceDate'] <= '2016/12/31')

In [31]:
dfInvoices.groupby(['VendorNumber','VendorName']).agg({'Freight':['sum']}).sort_values(('Freight', 'sum'), ascending=False).head(10)

Freight
                                                sum
VendorNumber VendorName                            
3960         DIAGEO NORTH AMERICA INC    257,032.07
4425         MARTIGNETTI COMPANIES       144,719.92
12546        JIM BEAM BRANDS COMPANY     123,880.97
17035        PERNOD RICARD USA           123,780.22
480          BACARDI USA INC              89,286.27
1392         CONSTELLATION BRANDS INC     79,528.99
1128         BROWN-FORMAN CORP            68,601.68
9165         ULTRA BEVERAGE COMPANY LLP   68,054.70
3252         E & J GALLO WINERY           61,966.91
9552         M S WALKER INC               55,551.82

# Deliverable 4c

In [32]:
dfFreight = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/InvoicePurchases12312016.csv")

In [33]:
df2016Invoices = dfFreight[(dfFreight['InvoiceDate']>='2016-01-01') & (dfFreight['InvoiceDate']<='2016-12-31')]

In [34]:
df2016Invoices.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
3           480  BACARDI USA INC              2016-01-12      8106   
4           516  BANFI PRODUCTS CORP          2016-01-07      8170   

       PODate     PayDate  Quantity    Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6     214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15     140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5     106.60     4.61      NaN  
3  2015-12-20  2016-02-05     10100 137,483.78 2,935.20      NaN  
4  2015-12-24  2016-02-12      1935  15,527.25   429.20      NaN

In [35]:
dfLargeVendors = df2016Invoices.groupby(['VendorNumber','VendorName'])[['Freight','Dollars','Quantity']].sum()

In [36]:
dfLargeVendors = dfLargeVendors[dfLargeVendors['Dollars']>=250000]

In [37]:
dfLargeVendors.head()

Freight       Dollars  Quantity
VendorNumber VendorName                                                   
480          BACARDI USA INC             86,153.58 16,985,053.64   1365360
516          BANFI PRODUCTS CORP          8,034.30  1,529,676.79    213967
653          STATE WINE & SPIRITS         7,757.25  1,481,052.81    149590
660          SAZERAC NORTH AMERICA INC.  17,198.00  3,387,877.78    481365
1128         BROWN-FORMAN CORP           66,576.22 13,139,921.12    973097

In [38]:
dfLargeVendors['FreightRate(%)'] = (dfLargeVendors['Freight']/dfLargeVendors['Dollars']) * 100
dfLargeVendors['Freightperunit'] = (dfLargeVendors['Freight']/dfLargeVendors['Quantity'])
dfLargeVendors.head()

Freight       Dollars  Quantity  \
VendorNumber VendorName                                                      
480          BACARDI USA INC             86,153.58 16,985,053.64   1365360   
516          BANFI PRODUCTS CORP          8,034.30  1,529,676.79    213967   
653          STATE WINE & SPIRITS         7,757.25  1,481,052.81    149590   
660          SAZERAC NORTH AMERICA INC.  17,198.00  3,387,877.78    481365   
1128         BROWN-FORMAN CORP           66,576.22 13,139,921.12    973097   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
480          BACARDI USA INC                        0.51            0.06  
516          BANFI PRODUCTS CORP                    0.53            0.04  
653          STATE WINE & SPIRITS                   0.52            0.05  
660          SAZERAC NORTH AMERICA INC.             0.51            0.04  
1128         BROWN-FORMAN CORP                      0.51            0.07

In [39]:
dfLargeVendors.sort_values(by='FreightRate(%)', ascending=True).head()

Freight      Dollars  Quantity  \
VendorNumber VendorName                                                     
6359         OLE SMOKY DISTILLERY LLC     1,876.61   377,754.50    108698   
9819         TREASURY WINE ESTATES       14,470.47 2,903,971.85    487654   
3089         SIDNEY FRANK IMPORTING CO    8,258.10 1,656,429.63    179772   
3924         HEAVEN HILL DISTILLERIES    13,854.16 2,773,519.92    346500   
8352         LUXCO INC                    9,785.35 1,951,124.00    271443   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
6359         OLE SMOKY DISTILLERY LLC               0.50            0.02  
9819         TREASURY WINE ESTATES                  0.50            0.03  
3089         SIDNEY FRANK IMPORTING CO              0.50            0.05  
3924         HEAVEN HILL DISTILLERIES               0.50            0.04  
8352         LUXCO INC                              0.50            0.04

In [40]:
dfLargeVendors.sort_values(by='FreightRate(%)', ascending=False).head(10)

Freight       Dollars  Quantity  \
VendorNumber VendorName                                                       
9625         WESTERN SPIRITS BEVERAGE CO   1,746.69    323,948.73     52243   
1590         DIAGEO CHATEAU ESTATE WINES   7,083.48  1,331,574.55    183317   
9744         FREDERICK WILDMAN & SONS      3,885.16    736,944.55     68087   
17031        FLAG HILL WINERY & VINEYARD   1,477.16    281,173.13     19188   
516          BANFI PRODUCTS CORP           8,034.30  1,529,676.79    213967   
653          STATE WINE & SPIRITS          7,757.25  1,481,052.81    149590   
4425         MARTIGNETTI COMPANIES       139,767.62 26,810,799.70   2544805   
8892         TRINCHERO FAMILY ESTATES     11,225.72  2,175,075.75    340040   
8673         STE MICHELLE WINE ESTATES    15,507.50  3,007,382.16    411345   
9815         WINE GROUP INC               26,280.87  5,109,629.23    863638   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
9625         WESTERN SPIRITS BEVERAGE CO            0.54            0.03  
1590         DIAGEO CHATEAU ESTATE WINES            0.53            0.04  
9744         FREDERICK WILDMAN & SONS               0.53            0.06  
17031        FLAG HILL WINERY & VINEYARD            0.53            0.08  
516          BANFI PRODUCTS CORP                    0.53            0.04  
653          STATE WINE & SPIRITS                   0.52            0.05  
4425         MARTIGNETTI COMPANIES                  0.52            0.05  
8892         TRINCHERO FAMILY ESTATES               0.52            0.03  
8673         STE MICHELLE WINE ESTATES              0.52            0.04  
9815         WINE GROUP INC                         0.51            0.03

# Deliverable 4d

In [41]:
dfSmallExp = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/InvoicePurchases12312016.csv")

In [42]:
dfSmallExp.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   
2           388  ATLANTIC IMPORTING COMPANY   2016-01-09      8169   
3           480  BACARDI USA INC              2016-01-12      8106   
4           516  BANFI PRODUCTS CORP          2016-01-07      8170   

       PODate     PayDate  Quantity    Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6     214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15     140.55     8.57      NaN  
2  2015-12-24  2016-02-16         5     106.60     4.61      NaN  
3  2015-12-20  2016-02-05     10100 137,483.78 2,935.20      NaN  
4  2015-12-24  2016-02-12      1935  15,527.25   429.20      NaN

In [43]:
dfSmallExp = dfSmallExp[dfSmallExp['Freight'] >= 100]
dfSmallExp = dfSmallExp[dfSmallExp['Quantity'] <= 1000]
dfSmallExp.head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
11          1485  CASTLE BRANDS CORP.          2016-01-08      8152   
16          2242  DELICATO VINEYARDS INC       2016-01-06      8139   
19          2555  DISARONNO INTERNATIONAL LLC  2016-01-11      8192   
23          2561  EDRINGTON AMERICAS           2016-01-08      8175   
28          3924  HEAVEN HILL DISTILLERIES     2016-01-08      8155   

        PODate     PayDate  Quantity  Dollars  Freight Approval  
11  2015-12-23  2016-02-19       320 5,420.41   179.26      NaN  
16  2015-12-22  2016-02-10       808 6,646.46   127.05      NaN  
19  2015-12-25  2016-02-17       385 3,506.41   146.86      NaN  
23  2015-12-24  2016-02-10       136 5,645.24   218.18      NaN  
28  2015-12-23  2016-02-15       818 7,079.02   200.02      NaN

In [44]:
dfSmallExp = dfSmallExp[['VendorNumber','VendorName','Freight','Dollars','Quantity']].groupby(['VendorNumber','VendorName']).sum().sort_values(by='Freight', ascending=False)
dfSmallExp.head()

Freight    Dollars  Quantity
VendorNumber VendorName                                               
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120
8664         STOLI GROUP,(USA) LLC         349.02  14,299.98       975
7239         REMY COINTREAU USA INC        348.07   7,515.38       382
653          STATE WINE & SPIRITS          303.70   5,544.89       541
6785         PALM BAY INTERNATIONAL INC    242.73   5,533.18       715

# Deliverable 4e

In [45]:
dfSmallExp['FreightRate(%)'] = (dfSmallExp['Freight']/dfSmallExp['Dollars']) * 100
dfSmallExp['Freightperunit'] = (dfSmallExp['Freight']/dfSmallExp['Quantity'])
dfSmallExp.head()

Freight    Dollars  Quantity  \
VendorNumber VendorName                                                  
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120   
8664         STOLI GROUP,(USA) LLC         349.02  14,299.98       975   
7239         REMY COINTREAU USA INC        348.07   7,515.38       382   
653          STATE WINE & SPIRITS          303.70   5,544.89       541   
6785         PALM BAY INTERNATIONAL INC    242.73   5,533.18       715   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
2561         EDRINGTON AMERICAS                     0.52            0.16  
8664         STOLI GROUP,(USA) LLC                  2.44            0.36  
7239         REMY COINTREAU USA INC                 4.63            0.91  
653          STATE WINE & SPIRITS                   5.48            0.56  
6785         PALM BAY INTERNATIONAL INC             4.39            0.34

In [46]:
dfSmallExp.sort_values(by='FreightRate(%)', ascending=False).head()

Freight  Dollars  Quantity  \
VendorNumber VendorName                                                
653          STATE WINE & SPIRITS          303.70 5,544.89       541   
7239         REMY COINTREAU USA INC        348.07 7,515.38       382   
6785         PALM BAY INTERNATIONAL INC    242.73 5,533.18       715   
2555         DISARONNO INTERNATIONAL LLC   146.86 3,506.41       385   
6213         MHW LTD                       177.50 4,821.83       412   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
653          STATE WINE & SPIRITS                   5.48            0.56  
7239         REMY COINTREAU USA INC                 4.63            0.91  
6785         PALM BAY INTERNATIONAL INC             4.39            0.34  
2555         DISARONNO INTERNATIONAL LLC            4.19            0.38  
6213         MHW LTD                                3.68            0.43

In [47]:
dfSmallExp.sort_values(by='FreightRate(%)', ascending=True).head(5)

Freight    Dollars  Quantity  \
VendorNumber VendorName                                                  
98450        Serralles Usa LLC             122.24  27,163.64       733   
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120   
4848         LAIRD & CO                    100.37  18,937.82       982   
1273         CALEDONIA SPIRITS INC         109.80  20,334.21       576   
2242         DELICATO VINEYARDS INC        127.05   6,646.46       808   

                                          FreightRate(%)  Freightperunit  
VendorNumber VendorName                                                   
98450        Serralles Usa LLC                      0.45            0.17  
2561         EDRINGTON AMERICAS                     0.52            0.16  
4848         LAIRD & CO                             0.53            0.10  
1273         CALEDONIA SPIRITS INC                  0.54            0.19  
2242         DELICATO VINEYARDS INC                 1.91            0.16

# Friday help session - question excused

In [66]:
pd.DataFrame([['Freight % of $',dfSmallExp['FreightRate(%)'].mean(),
               dfLargeVendors['FreightRate(%)'].mean(),df2016Invoices['FreightRate(%)'].mean(),
               ['Freight per Unit', dfSmallExp['Freightperunit'].mean()],
                dfLargeVendors['Freightperunit'].mean(),
                df2016Invoices['Freightperunit'].mean()]],
              columns=['', 'Small/Expensive','Large Vendors','All Purchases']
).set_index('')

KeyError: 'FreightRate(%)'

### The results from part e have much higher mzximum freight rate percentages than those in part c.

# Deliverable 5

In [48]:
dfSales = pd.read_csv("C:/Users/ryant/Desktop/Univ/Year 3/Spring 2024/ACTG 379/JN/SimplifiedData/HW4 Sampled Data/December OnlyPurchases_Sales/salesDecember2016Sample.csv")

In [49]:
dfSales.head(2)

InventoryId  Store  Brand                   Description   Size  \
0  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   
1  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   

   SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0              5        114.95       22.99  2016-12-01  750.00   
1              4         91.96       22.99  2016-12-03  750.00   

   Classification  ExciseTax  VendorNo                VendorName  
0               1       3.94      3960  DIAGEO NORTH AMERICA INC  
1               1       3.15      3960  DIAGEO NORTH AMERICA INC

In [50]:
begInvm = begInv[['InventoryId','Price']]

In [51]:
dfSales2 = pd.merge(left=dfSales, right=begInvm, how='inner', on=['InventoryId'])
dfSales2.head()

InventoryId  Store  Brand                  Description   Size  \
0  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
1  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
2  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
3  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
4  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   

   SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0              1         12.95       12.95  2016-12-01  750.00   
1              2         25.90       12.95  2016-12-02  750.00   
2              2         25.90       12.95  2016-12-03  750.00   
3              1         14.99       14.99  2016-12-13  750.00   
4              1         14.99       14.99  2016-12-17  750.00   

   Classification  ExciseTax  VendorNo                  VendorName  Price  
0               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99  
1               2       0.22      2000  SOUTHERN WINE & SPIRITS NE  14.99  
2               2       0.22      2000  SOUTHERN WINE & SPIRITS NE  14.99  
3               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99  
4               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99

In [52]:
dfSales2['CM'] = (dfSales2['SalesPrice']-dfSales2['Price'])
dfSales2.head()

InventoryId  Store  Brand                  Description   Size  \
0  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
1  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
2  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
3  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   
4  10_HORNSEY_10058     10  10058  F Coppola Dmd Ivry Cab Svgn  750mL   

   SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0              1         12.95       12.95  2016-12-01  750.00   
1              2         25.90       12.95  2016-12-02  750.00   
2              2         25.90       12.95  2016-12-03  750.00   
3              1         14.99       14.99  2016-12-13  750.00   
4              1         14.99       14.99  2016-12-17  750.00   

   Classification  ExciseTax  VendorNo                  VendorName  Price  \
0               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99   
1               2       0.22      2000  SOUTHERN WINE & SPIRITS NE  14.99   
2               2       0.22      2000  SOUTHERN WINE & SPIRITS NE  14.99   
3               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99   
4               2       0.11      2000  SOUTHERN WINE & SPIRITS NE  14.99   

     CM  
0 -2.04  
1 -2.04  
2 -2.04  
3  0.00  
4  0.00

In [57]:
dfSales3 = dfSales2.groupby(['InventoryId'])[['CM']].mean()
dfSales3.head()

CM
InventoryId           
10_HORNSEY_10058 -0.77
10_HORNSEY_10164  0.00
10_HORNSEY_1019   0.00
10_HORNSEY_10236  0.00
10_HORNSEY_10238  0.00

In [58]:
dfSales3.sort_values(by='CM', ascending=True).head()

CM
InventoryId              
69_MOUNTMEND_4652  -40.00
67_EANVERNESS_4652 -40.00
44_PORTHCRAWL_4652 -40.00
14_BROMWICH_4652   -40.00
32_MOUNTMEND_4652  -40.00

In [60]:
dfSales3.sort_values(by='CM', ascending=False).head(10)

CM
InventoryId              
33_HORNSEY_42884    40.00
69_MOUNTMEND_32900  30.00
73_DONCASTER_32900  30.00
33_HORNSEY_32900    30.00
34_PITMERDEN_32900  30.00
38_GOULCREST_32900  30.00
67_EANVERNESS_13949 26.00
66_EANVERNESS_13949 26.00
76_DONCASTER_13949  26.00
16_LUNDY_29431      25.00

### From the code I wrote above, I have found out that the 69_MOUNTMEND_4652 is one of the products with the lowest average contribution margin at -$40, and 33_HORNSEY_42884 was the product with the highest average contribution margin at \\\$40.